puffin/src/lib.rs

```diff
@@ -623,16 +775,18 @@ pub fn type_name_of<T>(_: T) -> &'static str {
 macro_rules! current_function_name {
     () => {{
         fn f() {}
-        let name = $crate::type_name_of(f);
-        // Remove "::f" from the name:
-        let name = &name.get(..name.len() - 3).unwrap();
-        $crate::clean_function_name(name)
+        $crate::type_name_of(f)
     }};
 }
 
 #[doc(hidden)]
 #[inline(never)]
-pub fn clean_function_name(name: &'static str) -> String {
+pub fn clean_function_name(name: &str) -> String {
+    let Some(name) = name.strip_suffix(USELESS_SCOPE_NAME_POSTFIX) else {
```

Comment:
nit: we should use the standard name (which the stdlib uses)

```suggestion
    let Some(name) = name.strip_suffix(USELESS_SCOPE_NAME_SUFFIX) else {
```